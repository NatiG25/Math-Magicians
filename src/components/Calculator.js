import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [obj, setObj] = useState({ total: 0, operation: null, next: null });

  function calculationHandler(e) {
    setObj(calculate(obj, e.target.textContent));
  }

  return (
    <>
      <header className="header">
        <h1>Math Magicians</h1>
        <nav>
          <Link to="/about">Quote</Link>
          <Link to="/">Home</Link>
          <Link to="/Calculator">Calculator</Link>
        </nav>
      </header>

      <div className="flex-container">
        <h2>Let us do some maths!</h2>
        <div className="grid-container">
          <div className="span-1">
            {obj.total}
            {obj.operation}
            {obj.next}
          </div>

          <button type="button" onClick={calculationHandler}>
            AC
          </button>
          <button type="button" onClick={calculationHandler}>
            +/-
          </button>
          <button type="button" onClick={calculationHandler}>
            %
          </button>
          <button
            type="button"
            className="operator"
            onClick={calculationHandler}
          >
            รท
          </button>

          <button type="button" onClick={calculationHandler}>
            7
          </button>
          <button type="button" onClick={calculationHandler}>
            8
          </button>
          <button type="button" onClick={calculationHandler}>
            9
          </button>
          <button
            type="button"
            className="operator"
            onClick={calculationHandler}
          >
            x
          </button>

          <button type="button" onClick={calculationHandler}>
            4
          </button>
          <button type="button" onClick={calculationHandler}>
            5
          </button>
          <button type="button" onClick={calculationHandler}>
            6
          </button>
          <button
            type="button"
            className="operator"
            onClick={calculationHandler}
          >
            -
          </button>

          <button type="button" onClick={calculationHandler}>
            1
          </button>
          <button type="button" onClick={calculationHandler}>
            2
          </button>
          <button type="button" onClick={calculationHandler}>
            3
          </button>
          <button
            type="button"
            className="operator"
            onClick={calculationHandler}
          >
            +
          </button>

          <button type="button" className="span-2" onClick={calculationHandler}>
            0
          </button>
          <button type="button" onClick={calculationHandler}>
            .
          </button>
          <button
            type="button"
            className="operator"
            onClick={calculationHandler}
          >
            =
          </button>
        </div>
      </div>
    </>
  );
};

export default Calculator;
