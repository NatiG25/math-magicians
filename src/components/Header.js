import React from 'react';
import { Link } from 'react-router-dom';

export default function Header() {
  return (
    <header className="header">
      <h1>Math Magicians</h1>
      <nav>
        <Link to="/quote">Quote</Link>
        <Link to="/">Home</Link>
        <Link to="/Calculator">Calculator</Link>
      </nav>
    </header>
  );
}
