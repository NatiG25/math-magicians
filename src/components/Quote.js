import React, { useEffect, useState } from 'react';

// Import images
import quoteLogo from '../images/quote-logo.png';
import study from '../images/study.jpeg';

export default function Quote() {
  const [quoteArrNum, setQuoteId] = useState(0);
  const [quoteApi, setQuoteApi] = useState('');
  const [authorApi, setAuthorApi] = useState('');

  const loader = document.querySelector('#loading');
  const quoteContent = document.querySelector('.quote-content');
  const displayLoading = () => {
    if (loader) {
      loader.classList.add('display');
      quoteContent.classList.add('hide-quote');
    }
  };

  const hideLoading = () => {
    if (loader) {
      quoteContent.classList.remove('hide-quote');
      loader.classList.remove('display');
    }
  };

  const apiUrl = 'https://mathematics-quotes-api.onrender.com/api/v1/quotes';

  const fetchQuote = async () => {
    displayLoading();
    try {
      const getQuote = await fetch(apiUrl);
      const res = await getQuote.json();
      const { quote, author } = res;
      hideLoading();
      setAuthorApi(author.toUpperCase());
      return setQuoteApi(quote.toUpperCase());
    } catch (error) {
      setAuthorApi(null);
      return setQuoteApi(null);
    }
  };

  useEffect(() => {
    fetchQuote();
  }, []);

  const quotes = [
    {
      id: 1,
      description: `MATHEMATICS IS NOT ABOUT NUMBERS, EQUATIONS, COMPUTATIONS, OR
      ALGORITHMS: IT IS ABOUT UNDERSTANDING.`,
      mathematician: 'WILLIAM PAUL THURSTON',
    },
    {
      id: 2,
      description:
        'PURE MATHEMATICS IS, IN ITS WAY, THE POETRY OF LOGICAL IDEAS.',
      mathematician: 'ALBERT EINSTEIN',
    },
    {
      id: 3,
      description:
        `THERE ARE TWO WAYS TO DO GREAT MATHEMATICS. THE FIRST IS TO BE SMARTER THAN EVERYBODY ELSE.
        THE SECOND WAY IS TO BE STUPIDER THAN EVERYBODY ELSE BUT PERSISTENT.`,
      mathematician: 'RAOUL BOTT',
    },
    {
      id: 4,
      description:
        'MATHEMATICS IS NOT A CAREFUL MARCH DOWN A WELL-CLEARED HIGHWAY, BUT A JOURNEY INTO A STRANGE WILDERNESS, WHERE THE EXPLORERS OFTEN GET LOST.',
      mathematician: 'RONALD GRAHAM',
    },
    {
      id: 5,
      description:
        'THE STUDY OF MATHEMATICS, LIKE THE NILE, BEGINS IN MINUTENESS BUT ENDS IN MAGNIFICENCE.',
      mathematician: 'CHARLES CALEB COLTON',
    },
    {
      id: 6,
      description:
        'MATHEMATICS IS THE QUEEN OF SCIENCES AND ARITHMETIC IS THE QUEEN OF MATHEMATICS.',
      mathematician: 'CARL FRIEDRICH GAUSS',
    },
  ];

  const changeQuote = () => {
    if (quoteApi && authorApi) fetchQuote();
    else setQuoteId(quoteArrNum < 5 ? quoteArrNum + 1 : 0);
  };

  return (
    <>
      <ul className="quote-list">
        <li key={quotes[quoteArrNum].id} className="quote-item">
          <div className="quote-container">
            <div className="quote-logo">
              <img src={quoteLogo} alt="quote logo" />
            </div>
            <div id="loading">Loading...</div>
            <div className="quote-content">
              <p className="quote-description">
                “
                {quoteApi || quotes[quoteArrNum].description}
                ”
              </p>
              <div className="author">
                <span>__</span>
                <span>{authorApi || quotes[quoteArrNum].mathematician}</span>
              </div>
            </div>
          </div>
        </li>
        <li className="new-quote-btn">
          <button type="button" onClick={changeQuote}>New quote</button>
        </li>
      </ul>
      <img src={study} alt="study" className="study-img" />
    </>
  );
}
