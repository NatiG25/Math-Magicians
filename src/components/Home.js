import React from 'react';
import Header from './Header';

export default function Home() {
  return (
    <>
      <main>
        <Header />
        <div className="home-content">
          <h2>Welcome to our page!</h2>
          <p>
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Minus
            laudantium maiores atque, iure quasi et illum odio est. Fugiat vitae
            possimus perferendis laudantium quae quam ipsum eligendi nam provident
            reiciendis.
          </p>
          <p>
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Minus
            laudantium maiores atque, iure quasi et illum odio est. Fugiat vitae
            possimus perferendis laudantium quae quam.
          </p>
        </div>

      </main>
    </>
  );
}
