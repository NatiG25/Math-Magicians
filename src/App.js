import React from 'react';
import { Routes, Route } from 'react-router-dom';
import Calculator from './components/Calculator';
import About from './components/About';
import Home from './components/Home';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Routes>
        <Route path="/about" element={<About />} />
        <Route path="/Calculator" element={<Calculator />} />
        <Route path="/" element={<Home />} />
      </Routes>
    );
  }
}

export default App;
