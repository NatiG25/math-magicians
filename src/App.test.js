import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import operate from './logic/operate';
import Calculator from './components/Calculator';
import Quote from './components/Quote';
import Header from './components/Header';
import Home from './components/Home';
import App from './App';
import calculate from './logic/calculate';

describe('Operations', () => {
  test('add two numbers', () => {
    expect(operate('25', '4', '+')).toBe('29');
  });

  test('Subtract two numbers', () => {
    expect(operate('25', '4', '-')).toBe('21');
  });

  test('Multiply two numbers', () => {
    expect(operate('25', '4', 'x')).toBe('100');
  });

  test('Divide two numbers', () => {
    expect(operate('25', '4', 'รท')).toBe('6.25');
  });
});

describe('render react components', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <App />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Functions test', () => {
  test('calculate.js test', () => {
    const initialItems = {
      total: 0,
      next: null,
      operation: null,
    };
    const buttonName = '=';
    if (buttonName === '.') {
      expect(calculate(initialItems, buttonName)).toEqual({
        total: 0,
        next: null,
        operation: null,
      });
    } else if (buttonName === '.') {
      expect(calculate(initialItems, buttonName)).toEqual({
        total: 0.0,
        next: null,
        operation: null,
      });
    } else if (buttonName === '1') {
      expect(calculate(initialItems, buttonName)).toEqual({
        total: null,
        next: '1',
      });
    } else if (buttonName === '=') {
      expect(calculate(initialItems, buttonName)).toEqual({});
    }
  });
});
