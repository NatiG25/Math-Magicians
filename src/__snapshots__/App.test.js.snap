// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render react components renders correctly 1`] = `
<div
  className="flex-container"
>
  <h2>
    Let us do some maths!
  </h2>
  <div
    className="grid-container"
  >
    <div
      className="span-1"
    >
      0
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="span-2"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`render react components renders correctly 2`] = `
<main>
  <p
    className="quote"
  >
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding --Willam Paul Thurston
  </p>
</main>
`;

exports[`render react components renders correctly 3`] = `
<main>
  <div
    className="home-content"
  >
    <h2>
      Welcome to our page!
    </h2>
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Minus laudantium maiores atque, iure quasi et illum odio est. Fugiat vitae possimus perferendis laudantium quae quam ipsum eligendi nam provident reiciendis.
    </p>
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Minus laudantium maiores atque, iure quasi et illum odio est. Fugiat vitae possimus perferendis laudantium quae quam.
    </p>
  </div>
</main>
`;

exports[`render react components renders correctly 4`] = `
<header
  className="header"
>
  <h1>
    Math Magicians
  </h1>
  <nav>
    <a
      href="/about"
      onClick={[Function]}
    >
      Quote
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/Calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
  </nav>
</header>
`;

exports[`render react components renders correctly 5`] = `
Array [
  <header
    className="header"
  >
    <h1>
      Math Magicians
    </h1>
    <nav>
      <a
        href="/about"
        onClick={[Function]}
      >
        Quote
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </nav>
  </header>,
  <main>
    <div
      className="home-content"
    >
      <h2>
        Welcome to our page!
      </h2>
      <p>
        Lorem ipsum, dolor sit amet consectetur adipisicing elit. Minus laudantium maiores atque, iure quasi et illum odio est. Fugiat vitae possimus perferendis laudantium quae quam ipsum eligendi nam provident reiciendis.
      </p>
      <p>
        Lorem ipsum, dolor sit amet consectetur adipisicing elit. Minus laudantium maiores atque, iure quasi et illum odio est. Fugiat vitae possimus perferendis laudantium quae quam.
      </p>
    </div>
  </main>,
]
`;
